### DISCUSSION TITLE
getSearchableModules

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#15181] [How to] [answered]
Possibly dump question, but... how to determine if my module (custom) falling under getSearchableModules?
Got error in custom function: 
: Uncaught Error: Call to a member function hasModuleActionPermission() on null in /var/www/html/crm/app/RecordSearch.php:345 Stack trace: #0 /var/www/html/crm/app/Record.php(236): App\RecordSearch::getSearchableModules() #1 /var/www/html/crm/modules/Vtiger/models/Record.php(546): App\Record::updateLabelOnSave(Object(Vtiger_Record_Model)) #2 /var/www/html/crm/my_function.php(54): Vtiger_Record_Model->save() #3 {main} thrown in
/var/www/html/crm/app/RecordSearch.php

**Comment by rskrzypczak** [comment, marked as answer]
You don't have a problem with global search permissions here, but with the user.
Presumably, you did not declare a user in your script (my_function.php)
Enter at the beginning of the script, e.g.
`\App\User::setCurrentUserId(YOUR_USER_ID);` [(example)](https://github.com/YetiForceCompany/YetiForceScripts/blob/0dd92ca44fa5f9db0f50480f5fae2d515c61765b/Data%20migration/PhoneParser.php#L14)
This should solve your problem.
